### DISCUSSION TITLE
How can I bulk import osu!stable installation on an existing osu!lazer installation

### DISCUSSION THREAD

**Discussion by alexpado** [ppy/osu#27800] [Q&A] [answered]
**For context:** With some friend, we will have a booth to present the game at a convention. We choose to present the game using osu!lazer this year. Every year, we share our beatmaps folder (and skins) to have the same library on each computer used on the booth.

I've been using osu!lazer for quite some time now, without having osu!stable installed. This mean that my beatmaps aren't hardlinked.

1. How can I import a songs folder without loosing my own library ?
2. How can I massively export my library so my friend can import those on their "hardlinked" installs ? (doing it one by one over 40,000+ maps isn't realistic)

For the 1st question, I though about a script which bundle each folder into a .osz file, then dropping everything on the game, but there might be a better solution. I have absolutely 0 idea for the 2nd question though

**Comment by SupDos** [comment]
You can run the stable importer within lazer at any point, so if you have a stable installation you can import it, and it will add them alongside your current maps on lazer

Though for your situation the easiest thing to do would be to get the entire `osu` folder (the one that has `files`, `client.realm`, etc) and just copy it over to the other computer, that way it will have the exact same beatmaps, skins, settings etc. Make sure you log out of lazer before doing this.

**Reply by alexpado** [reply]
Ok so if I understood what you said, I need to install osu!stable, copy the song folder in the install folder, run the import and then uninstall it.

Then once I have done that, logout on lazer, copy my lazer folder on each computer to 100% sync every install.

**Reply by SupDos** [reply, marked as answer]
If you don't have stable installed, you can select the folder where you have the `Songs` folder on its own without having to install stable and drag it into there first.
![image](https://github.com/ppy/osu/assets/6813986/23dcc989-f17a-43c8-b33d-c3986ba5e0fd)
If it doesn't work for some reason though, then doing it through stable as you said will be fine.

Logging out is mainly so that the other computer doesn't also log you in automatically, since the lazer `osu` folder contains your login token.
